python/variable_init.cpp

```diff
@@ -205,21 +206,16 @@ auto make_element_array(const Dimensions &dims, const py::object &source,
 template <class T> struct MakeVariable {
   static Variable apply(const Dimensions &dims, const py::object &values,
                         const py::object &variances, const units::Unit unit) {
-    const auto [actual_unit, conversion_factor] = common_unit<T>(values, unit);
-    if (conversion_factor != 1) {
-      // TODO Triggered once common_unit implements conversions.
-      std::terminate();
-    }
-
+    const auto [values_unit, final_unit] = common_unit<T>(values, unit);
     auto values_array =
-        Values(make_element_array<T>(dims, values, actual_unit));
+        Values(make_element_array<T>(dims, values, values_unit));
     auto variable = variances.is_none()
                         ? makeVariable<T>(dims, std::move(values_array))
                         : makeVariable<T>(dims, std::move(values_array),
                                           Variances(make_element_array<T>(
-                                              dims, variances, actual_unit)));
-    variable.setUnit(actual_unit);
-    return variable;
+                                              dims, variances, values_unit)));
+    variable.setUnit(values_unit);
+    return to_unit(variable, final_unit);
```

Comment:
This relies on the fact that `to_unit` does nothing if unit is unchanged?

Reply:
Yes.
Would you rather check again to be sure we don't copy in case `to_unit` changes?

Reply:
No. If anything introduce the optional `copy` argument for `to_unit` that I had planned:

```c++
Variable to_unit(Variable, Unit, bool copy=false);
```
and here:
```c++
return to_unit(variable, final_unit, false);
```
(false is default, but we can be explicit here for doc purposes and future proofing).

Reply:
Better: Make an enum, this would be used in more places in the future:
```python
enum class CopyPolicy { }; // good names wanted
```

Reply:
See #2016

Reply:
Waiting here until we settled on a default, so you can use the correct arg here, i.e., merge this after #2016?